src/main/java/life/qbic/domain/sample/Sample.java

```diff
@@ -104,20 +106,26 @@ public <T extends SampleEvent> void handle(T event) {
     if (events.contains(event)) {
       return;
     }
-    if (!occurredAfterCurrentState(event)) {
-      throw new UnrecoverableException(
-          String.format("The sample (%s) was modified after %s", sampleCode, event.occurredOn()));
-    }
+    validateEventTimeOrThrowException(event);
     apply(event);
     events.add(event);
   }
 
-  private boolean occurredAfterCurrentState(SampleEvent event) {
+  private void validateEventTimeOrThrowException(SampleEvent event) {
```

Comment:
NIT: a little verbose, having the exception in the signature and docs is sufficient imo

Reply:
I included the possibility of exception in the method name so by reading the code it is apparent when the process can fail.
```
    validateEventTimeOrThrowException(event);
    apply(event);
    events.add(event);
```
I can change that if you think that it is self-explanatory.